gnocchi/cli/metricd.py

```diff
@@ -126,6 +127,11 @@ def __init__(self, worker_id, conf):
         # This stores the last time the processor did a scan on all the sack it
         # is responsible for
         self._last_full_sack_scan = utils.StopWatch().start()
+        # Only update the list of sacks to process every
+        # metric_processing_delay
+        self._get_sacks_to_process = cachetools.func.ttl_cache(
+            ttl=conf.metricd.metric_processing_delay
```

Comment:
we use to use worker_sync_rate to control this? we want to reuse it? i don't know if we want to force non-notifier drivers to have to sync every time?

https://github.com/gnocchixyz/gnocchi/blob/stable/4.0/gnocchi/cli.py#L212

Reply:
We removed `worker_sync_rate` to simplify the configuration and I think it's good that way. Keeping things simple to configure. You wont your worker to be in sync each time you process, so every `metric_processing_delay` seconds. I don't see the point to add yet another knob.

I don't understand your point of non-notifier though. It should not change anything since `_get_sack_to_process` will be called every `metric_processing_delay` so the cache will be invalidated each time – so no cache in the end and called every `metric_processing_delay` as  before.

Reply:
right, but previously, if i put a low metric_processing_delay to simulate 'real-time', i could still leave the run_watchers to just redistribute tasks at a less aggressive interval.

but i see i approved the original commit so meh.